### PR TITLE
UAVCANv1 port unset fixes and deinitialization

### PR DESCRIPTION
**Pull request by PetervdPerk-NXP**
UAVCANv1
Fixes handling unset caused by #17733 and correct deinitialization by freeing correct objects.

### CODE REVIEW

File: src/drivers/uavcan_v1/Subscribers/DynamicPortSubscriber.hpp [hunk 1]
```diff
@@ -70,13 +70,7 @@ class UavcanDynamicPortSubscriber : public UavcanBaseSubscriber
...
 
 			if (_param_manager.GetParamByName(uavcan_param, value)) {
-				int32_t new_id = value.integer32.value.elements[0];
-
-				// Allow, for example, a default PX4 param value of '-1' to disable subscription
-				if (!isValidPortId(new_id)) {
-					// but always use the standard 'unset' value for comparison
-					new_id = CANARD_PORT_ID_UNSET;
-				}
+				uint16_t new_id = value.natural16.value.elements[0];
```
Comment (JacobCrabill) [inline src/drivers/uavcan_v1/Subscribers/DynamicPortSubscriber.hpp:73]: Don't we still need a check after here to ensure that `new_id <= CANARD_PORT_ID_MAX`?

Comment (PetervdPerk) [reply src/drivers/uavcan_v1/Subscribers/DynamicPortSubscriber.hpp:73]: The callback return a boolean, my assumption would be that if it's returning true it means it got a valid port_id >=0 and <= CANARD_PORT_ID_MAX

Comment (JacobCrabill) [reply src/drivers/uavcan_v1/Subscribers/DynamicPortSubscriber.hpp:73]: Oh, I see. 👍🏻

Comment (PetervdPerk-NXP) [reply src/drivers/uavcan_v1/Subscribers/DynamicPortSubscriber.hpp:73]: I think this is ready for merge now.

**Review by dagar — APPROVED** [review]

**Pull request merged**